src/server/svr_jobfunc.c

```diff
@@ -4306,6 +4327,10 @@ start_end_dur_wall(resc_resv *presv)
 					prsc->rs_value.at_val.at_long))
 				rc = -1;
 			else {
+				if (pstate == RESV_BEING_ALTERED) {
+					presv->ri_alter.ra_flags |= RESV_DURATION_MODIFIED;
+					presv->ri_alter.ra_duration = pduration->at_val.at_long;
+				}
```

Comment:
Any time you change duration, you need to change walltime as well.

Reply:
we don't store walltime in the resv_alter struct, it's only for when we need to revert